feat(auth): redirect to `targetPath` after login

InosRahul opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes  https://github.com/langfuse/langfuse/issues/1093

 Loom Video: https://www.loom.com/share/a5da2b975d3c42f7a2c7e63cdcc102b6?sid=a5a83a36-3fa3-4f47-a242-b68335ca96bb


## Type of change

<!-- Please delete bullets that are not relevant. -->

- [x] New feature (non-breaking change which adds functionality)


## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

InosRahul committed:
feat(auth): redirect to target url after sign in

InosRahul committed:
test(auth): redirect user to target url after sign in

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/langfuse/langfuse?pullRequest=1913) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/langfuse/langfuse?pullRequest=1913) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/langfuse/langfuse?pullRequest=1913) it.</sub>

InosRahul committed:
Merge branch 'main' into feat/post_login_redirect_url

marcklingen commented on web/src/components/layouts/layout.tsx:
```diff
@@ -136,31 +136,43 @@ export default function Layout(props: PropsWithChildren) {
 
   const hideNavigation =
     session.status === "unauthenticated" ||
+    projects.length === 0 ||
```
this was removed in a recent change to main, please remove

InosRahul replied:
Done.

InosRahul replied:
@marcklingen is this okay or does it need more changes? Also I noticed the tests CI tests are failing for forked repositories, is this intended behaviour or do I need to change some config?

marcklingen replied:
we fixed the ci test, sorry for the confusion here

marcklingen reviewed (commented):
thanks for the contribution, this looks good mostly, can you remove this line?

InosRahul committed:
fix: remove projects.length check

marcklingen committed:
simplify and rename

marcklingen commented on web/src/components/layouts/layout.tsx:
```diff
@@ -144,18 +144,25 @@ export default function Layout(props: PropsWithChildren) {
   if (
     session.status === "unauthenticated" &&
     !unauthenticatedPaths.includes(router.pathname) &&
-    !publishablePaths.includes(router.pathname) &&
```
why did you remove this line?

marcklingen reviewed (commented):
made little changes (cosmetic), one last questions, otherwise seems good to go. thank you!

marcklingen committed:
Merge branch 'main' into pr/InosRahul/1913

marcklingen committed:
revert removing public pages

marcklingen committed:
include target path in sign in button on no access error pages

marcklingen commented on web/src/__e2e__/auth.spec.ts:
```diff
@@ -70,3 +70,54 @@ test("Signup input validation", async ({ page }) => {
+
+  await expect(page).toHaveURL(/.*traces/);
+
+  await page
```
this test seems to fail, can you fix it?

marcklingen reviewed (changes requested):
made additional changes, PR ready to merge but tests keep failing. Can you fix the end to end test?

InosRahul committed:
Merge branch 'langfuse:main' into feat/post_login_redirect_url

marcklingen commented:
maybe you can just try to navigate to /settings of the project without really needing to press any buttons beyond the login screen, assuming that this is the issue here

InosRahul committed:
test: remove clicking logic

InosRahul commented:
> maybe you can just try to navigate to /settings of the project without really needing to press any buttons beyond the login screen, assuming that this is the issue here

Fixed now, also cleaned up the commits.

marcklingen committed:
Merge branch 'main' into feat/post_login_redirect_url

marcklingen commented:
Looks great, the worker CI test fails on forks, we're about to fix this to make it easier to contribute to langfuse

marcklingen commented:
Thank you for your contribution!
